plugin/helpers.go

```diff
@@ -69,3 +76,24 @@ func (p *HelpersImpl) ensureServerVersion(required string) error {
 	}
 	return nil
 }
+
+func (p *HelpersImpl) InstallPluginFromUrl(downloadUrl string, replace bool) (*model.Manifest, error) {
+	parsedUrl, _ := url.Parse(downloadUrl)
+	if !*p.API.GetConfig().PluginSettings.AllowInsecureDownloadUrl && parsedUrl.Scheme != "https" {
+		return nil, errors.New("downloading from insecure url is not allowed")
+	}
+
+	client := &http.Client{Timeout: 60 * time.Minute}
+	response, err := client.Get(downloadUrl)
```

Comment:
Hey Rajat, can you please do some validation here by checking `response.Status` to see if we get a `20x` status code(checking for 200 and 201 should be enough)?

Other status codes should indicate an error and in that case, `response.Body` probably will stream nothing or an error message related data which shouldn't be consumed by the `InstallPlugin()`.